### PR TITLE
Support Microsoft Entra ID authentication and authorization for azure-storage-blob extension

### PR DESCRIPTION
**Pull request by majguo**
This pull request includes several changes to the Azure integration and documentation in the project. The most significant changes involve enhancements to the authentication methods for **Azure Storage Blob** service and updates to the documentation to reflect these changes.

### Codebase Updates:

* [`extensions/azure-storage-blob/runtime/src/main/java/io/quarkiverse/azure/storage/blob/runtime/StorageBlobConfig.java`](diffhunk://#diff-ef270aaccb8cd078c480b569730c9e73c8e44076940f02d134dc93808f8f47e5L21-R28): Added a new configuration property for the Azure Storage Blob endpoint.
* [`extensions/azure-storage-blob/deployment/src/main/java/io/quarkiverse/azure/storage/blob/deployment/DevServicesStorageBlobProcessor.java`](diffhunk://#diff-c1fc4fced87f25fb3efa03ebe1b248eb12adf8378cd3d8b1620ac9105eea1ff5L41-R42): Updated to start DevService only when neither `quarkus.azure.storage.blob.connection-string` nor `quarkus.azure.storage.blob.endpoint` is configured. [[1]](diffhunk://#diff-c1fc4fced87f25fb3efa03ebe1b248eb12adf8378cd3d8b1620ac9105eea1ff5L41-R42) [[2]](diffhunk://#diff-c1fc4fced87f25fb3efa03ebe1b248eb12adf8378cd3d8b1620ac9105eea1ff5L139-R141) [[3]](diffhunk://#diff-c1fc4fced87f25fb3efa03ebe1b248eb12adf8378cd3d8b1620ac9105eea1ff5L161-R171)

### Tests Enhancements:

* [`.github/run-integration-test.sh`](diffhunk://#diff-8c48f063e81b678b3cfc7bcc9db009ab6b7d49f189c756e0afe06c8d62e93c03L12-R52): Added Microsoft Entra ID authentication to Azure Storage Blob, and updated the script to run integration tests with both native executables and in JVM mode, and with both Microsoft Entra ID and connection string authentication. [[1]](diffhunk://#diff-8c48f063e81b678b3cfc7bcc9db009ab6b7d49f189c756e0afe06c8d62e93c03L12-R52) [[2]](diffhunk://#diff-8c48f063e81b678b3cfc7bcc9db009ab6b7d49f189c756e0afe06c8d62e93c03R64-L49) [[3]](diffhunk://#diff-8c48f063e81b678b3cfc7bcc9db009ab6b7d49f189c756e0afe06c8d62e93c03L61-R114)

### Documentation Updates:

* [`docs/modules/ROOT/pages/quarkus-azure-storage-blob.adoc`](diffhunk://#diff-f7ac435bd08d0fa2e209ceac03cd03a4f8a468728b8049bc58a9ad1ef48c04ddL68-R108): Added instructions for authenticating to Azure Storage Blob using Microsoft Entra ID and connection string, and updated the configuration reference section. [[1]](diffhunk://#diff-f7ac435bd08d0fa2e209ceac03cd03a4f8a468728b8049bc58a9ad1ef48c04ddL68-R108) [[2]](diffhunk://#diff-f7ac435bd08d0fa2e209ceac03cd03a4f8a468728b8049bc58a9ad1ef48c04ddR119-R121)
* [`docs/modules/ROOT/pages/quarkus-azure-cosmos.adoc`](diffhunk://#diff-025098588d9e30823b345507584ff18919a5e3022b62e090fce15425d702d970L60-L73): Updated the documentation to include instructions for authenticating to Azure Cosmos DB using Microsoft Entra ID and key-based authentication. [[1]](diffhunk://#diff-025098588d9e30823b345507584ff18919a5e3022b62e090fce15425d702d970L60-L73) [[2]](diffhunk://#diff-025098588d9e30823b345507584ff18919a5e3022b62e090fce15425d702d970L106-R127)

### Dependency Additions:

* `extensions/azure-storage-blob/deployment/pom.xml` and `extensions/azure-storage-blob/runtime/pom.xml`: Added a dependency on `quarkus-azure-identity` to support the new authentication methods. [[1]](diffhunk://#diff-6f1f7f5e60dd8fea6d9b36363361b2eda221b031378df7228164ebca1fc9c6edR19-R22) [[2]](diffhunk://#diff-a467bf963ffff4d27c038a863990a231eea097cd40ea17919c59d4e25ffbd4a1R19-R22)

These changes collectively enhance the project's capability to authenticate with Azure services more securely and flexibly, and improve the documentation to guide users through these new authentication methods.

Signed-off-by: Jianguo Ma <jiangma@microsoft.com>

### CODE REVIEW

File: .github/run-integration-test.sh [hunk 1]
```diff
@@ -9,67 +9,106 @@ set -Euo pipefail
...
+unset QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT
+mvn -f azure-storage-blob/pom.xml -B test-compile failsafe:integration-test -Dnative -Dazure.test=true
+
+# Run both unit test and integration test in JVM mode against Azure services
+export QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT=${AZURE_STORAGE_BLOB_ENDPOINT}
+unset QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING
+mvn -f azure-storage-blob/pom.xml -B verify -Dazure.test=true
+export QUARKUS_AZURE_STORAGE_BLOB_CONNECTION_STRING=${AZURE_STORAGE_BLOB_CONNECTION_STRING}
+unset QUARKUS_AZURE_STORAGE_BLOB_ENDPOINT
+mvn -f azure-storage-blob/pom.xml -B verify -Dazure.test=true
```
Comment (backwind1233) [inline .github/run-integration-test.sh:51]: how about 
```
mvn -f azure-storage-blob/pom.xml -B verify -Dazure.test=true -Dquarkus.azure.storage.blob.connection-string=${AZURE_STORAGE_BLOB_CONNECTION_STRING}
```

**Review by backwind1233 — APPROVED** [review]
LGTM

**Review by galiacheng — APPROVED** [review]
LGTM

File: .github/run-integration-test.sh (hunk 1 above)
Comment (majguo) [reply .github/run-integration-test.sh]: Nice suggestion, thanks! Fixed in 428f8edd575966a6fd77a9cc62bac89f3421d0f5.

**Review by galiacheng — APPROVED** [review]
LGTM, thank you @backwind1233

**Pull request merged**